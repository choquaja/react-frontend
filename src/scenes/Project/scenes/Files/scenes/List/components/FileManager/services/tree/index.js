import { compose } from 'redux';

export const sortDirFirst = (a, b) => {
  if (a.isDirectory !== b.isDirectory) {
    return a.isDirectory ? -1 : 1;
  }
  return a.parts[a.parts.length - 1] > b.parts[b.parts.length - 1] ? 1 : -1;
};

export const createDirectory = (parts, firstFile) => ({
  parts,
  name: parts[parts.length - 1],
  relPath: parts.join('/'),
  isDirectory: true,
  files: [firstFile],
});

export const groupByDirectory = depth => (map, file) => {
  // TODO: handle empty directories
  // Below will treat empty dirs like a file
  if (file.parts.length === (depth + 1)) {
    return map.set(file.relPath, file);
  }
  if (map.has(file.parts[depth])) {
    const directory = map.get(file.parts[depth]);
    directory.files = [...directory.files, file];
    return map.set(file.parts[depth], directory);
  }
  const parts = file.parts.slice(0, depth + 1);
  return map.set(file.parts[depth], createDirectory(parts, file));
};

export const recurseIfDir = depth => (file) => {
  if (!file.isDirectory) return file;
  return {
    ...file,
    files: reduceToFileTree(depth + 1)(file.files), // eslint-disable-line
  };
};

export const reduceToFileTree = depth => files => [
  ...files.reduce(groupByDirectory(depth), new Map()).values(),
].sort(sortDirFirst).map(recurseIfDir(depth));
export const startReducing = reduceToFileTree(0);
export const addName = file => ({ ...file, name: file.parts[file.parts.length - 1] });
export const addPathParts = file => ({ ...file, parts: file.relPath.split('/') });
export const createRelPath = file => ({
  ...file,
  relPath: file.path + file.name,
});
export const prepareFiles = files =>
  files.filter(Boolean).map(createRelPath).map(addPathParts).map(addName);
export const getFileTree = compose(startReducing, prepareFiles);
