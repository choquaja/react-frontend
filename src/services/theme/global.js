import { injectGlobal } from 'styled-components';
import normalize from 'polished/lib/mixins/normalize';
import { theme, themeFont, themeColor } from '.';

/* eslint-disable no-unused-expressions */
injectGlobal`
  ${normalize()}
  * {
    box-sizing: border-box;
  }
  *:before,
  *:after {
    box-sizing: border-box;
  }
  html,
  body,
  main {
    height: 100%;
  }
  html {
    font-size: 10px;
    -webkit-tap-highlight-color: rgba(0,0,0,0);
  }
  body {
    margin: 0;
    padding: 0;
    font-family: ${themeFont('body')({ theme })};
    font-size: 1.4rem;
    background-color: ${themeColor('backdrop')({ theme })};
    color: ${themeColor('tertiary')({ theme })};
  }
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-family: ${themeFont('heading')({ theme })};
  }
`;
/* eslint-enable */
