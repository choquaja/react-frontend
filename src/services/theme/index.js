import getOr from 'lodash/fp/getOr';

export const theme = {
  columns: {
    number: 12,
  },
  breakpoints: {
    sm: '40rem',
    md: '70rem',
    lg: '100rem',
    xl: '130rem',
  },
  colors: {
    primary: '#f47920',
    primaryVariant1: '#f7901d',
    primaryVariant2: '#fcb361',
    secondary: '#20c5f4',
    secondaryVariant1: '#83e9ff',
    secondaryVariant2: '#c6f0ff',
    tertiary: '#182752',
    tertiaryVariant1: '#485372',
    tertiaryVariant2: '#8a90a3',
    success: '#4caf50',
    info: '#2196f3',
    warning: '#9c27b0',
    error: '#e51c23',
    white: '#fff',
    gray1: '#f4f4f4',
    gray2: '#dfdfdf',
    gray3: '#bdbdbd',
    gray4: '#8c8c8c',
    gray5: '#5e5e5e',
    gray6: '#3b3b3b',
    black: '#111',
    backdrop: '#f2f7fa',
  },
  fonts: {
    body: "'Nunito Sans', sans-serif",
    heading: "'Roboto', sans-serif",
  },
};

/**
 * Gets value from theme object by path
 * @param  {String[]} paths - path to desired value as multiple arguments
 * @param  {Object} props - object with theme as a property
 * @return {*|undefined} - value at path
 */
export const themePart = (...paths) => getOr('')(['theme', ...paths]);

export const themeDefaultColumns = themePart('columns', 'number');

export const themeBreakpoints = themePart('breakpoints');
export const themeBreakpoint = bp => themePart('breakpoints', bp);

export const themeColors = themePart('colors');
export const themeColor = color => themePart('colors', color);

export const themeFonts = themePart('fonts');
export const themeFont = type => themePart('fonts', type);
