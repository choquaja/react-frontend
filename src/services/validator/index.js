import Validator from 'validatorjs';

Validator.register(
  'projectName',
  value => value && /^[\w-]+$/.test(value),
  'The :attribute can only contain letters, numbers, dashes, and underscores.',
);

export default function validator(rules = {}, customErrorMessages = {}) {
  const errorMessages = {
    required: 'This field is required.',
    ...customErrorMessages,
  };
  return (values) => {
    const instance = new Validator(values, rules, errorMessages);
    instance.passes();
    return instance.errors.all();
  };
}
