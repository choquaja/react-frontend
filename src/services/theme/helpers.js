import { css } from 'styled-components';
import borderRadius from 'polished/lib/shorthands/borderRadius';
import placeholder from 'polished/lib/mixins/placeholder';
import { themeColor } from '../theme';

export const groupedElement = ({ borderWidth, radiusWidth }) => css`
  box-shadow: none;
  border-radius: 0;
  border-left-width: 0;
  border-right-width: 0;
  &:first-child {
    border-left-width: ${borderWidth};
    ${borderRadius('left', radiusWidth)}
  }
  &:last-child {
    border-right-width: ${borderWidth};
    ${borderRadius('right', radiusWidth)}
  }
`;

export const formElementStyles = css`
  display: inline-block;
  ${props => props.block && css`display: block;`}
  width: 30rem;
  ${props => props.full && css`width: 100%;`}
  color: ${themeColor('tertiary')};
  background-color: ${themeColor('white')};
  border: 1px solid ${themeColor('gray2')};
  border-radius: 2px;
  padding: .7em;
  font-size: 1.6rem;
  transition: all .2s ease-in-out;
  ${props => props.large && css`
    font-size: 1.9rem;
    height: 4.2rem;
  `}
  ${props => props.small && css`
    font-size: 1.3rem;
    height: 2.6rem;
  `}
  ${props => props.camo && css`
    padding: .4rem;
    line-height: inherit;
    font-size: inherit;
    font-weight: inherit;
    color: inherit;
    border-color: transparent;
    &:hover {
      border-color: ${themeColor('gray2')};
    }
  `}
  &:disabled {
    cursor: default;
    background: ${themeColor('gray2')};
    opacity: .4;
  }
  ${placeholder(css`
    color: ${themeColor('gray3')};
    font-style: italic;
  `)}
`;
