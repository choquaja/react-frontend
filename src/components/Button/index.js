import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import darken from 'polished/lib/color/darken';
import timingFunctions from 'polished/lib/mixins/timingFunctions';
import { groupedElement } from '../../services/theme/helpers';

const darkenColor = scale => color => darken(scale, color);
const darkenHover = darkenColor(0.12);
const darkenActive = darkenColor(0.24);

const createButtonVariant = bg => css`
  color: white;
  background-color: ${bg};
  border-color: ${bg};
  &:hover:not(:disabled) {
    background-color: ${darkenHover(bg)};
    border-color: ${darkenHover(bg)};
  }
  &:active:not(:disabled) {
    background-color: ${darkenActive(bg)};
    border-color: ${darkenActive(bg)};
  }
`;

const Button = styled.button`
  display: inline-block;
  ${props => props.block && css`display: block;`}
  ${props => props.full && css`width: 100%;`}
  background-color: #e2e2e2;
  border: 1px solid #e2e2e2;
  border-radius: 2px;
  color: #333;
  cursor: pointer;
  font-family: inherit;
  font-size: 1.6rem;
  height: 3.6rem;
  ${props => props.large && css`
    font-size: 1.9rem;
    height: 4.2rem;
  `}
  ${props => props.small && css`
    font-size: 1.2rem;
    height: 2.6rem;
  `}
  font-weight: 700;
  letter-spacing: normal;
  ${''/* height: 40px; */}
  ${''/* line-height: 20px; */}
  padding: .5em 1.25em;
  text-transform: none;
  text-align: center;
  touch-action: manipulation;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;
  ${''/* box-shadow: rgba(0, 0, 0, 0.117647) 0px 1px 6px, rgba(0, 0, 0, 0.117647) 0px 1px 4px; */}
  box-shadow: 0 0 2px rgba(0,0,0,.12), 0 2px 2px rgba(0,0,0,.2);
  ${props => props.flat && css`box-shadow: none;`}
  transition: all .15s ${timingFunctions('easeIn')};
  &:hover:not(:disabled) {
    background-color: ${darkenHover('#c4c4c4')};
    border-color: ${darkenHover('#c4c4c4')};
  }
  &:active:not(:disabled) {
    background-color: ${darkenActive('#c4c4c4')};
    border-color: ${darkenActive('#c4c4c4')};
  }
  &:disabled {
    cursor: default;
    opacity: .4;
  }
  ${props => props.grouped && groupedElement({ borderWidth: '1px', radiusWidth: '2px' })}
  ${props => props.primary && createButtonVariant('#f2994a')}
  ${props => props.secondary && createButtonVariant('#886cff')}
  ${props => props.success && createButtonVariant('#7ae43b')}
  ${props => props.info && createButtonVariant('#1fa2f3')}
  ${props => props.warning && createButtonVariant('#ffbc01')}
  ${props => props.error && createButtonVariant('#ff3f80')}
`;

Button.propTypes = {
  children: PropTypes.node.isRequired,
  block: PropTypes.bool,
  full: PropTypes.bool,
  large: PropTypes.bool,
  small: PropTypes.bool,
  grouped: PropTypes.bool,
  primary: PropTypes.bool,
  secondary: PropTypes.bool,
  success: PropTypes.bool,
  info: PropTypes.bool,
  warning: PropTypes.bool,
  error: PropTypes.bool,
};

Button.defaultProps = {
  block: false,
  full: false,
  large: false,
  small: false,
  grouped: false,
  primary: false,
  secondary: false,
  success: false,
  info: false,
  warning: false,
  error: false,
};

export default Button;
