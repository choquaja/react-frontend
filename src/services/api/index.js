import axios from 'axios';
import { isLoggedIn, getToken } from '../authToken';
import callsList from './callsList';
import * as helpers from './helpers';

const instance = axios.create({
  baseURL: 'https://go-pilot.3blades.io/',
});

const prepareAxios = ({ history }) => call => (data, options = {}) => instance({
  ...options,
  data,
  method: call.method,
  url: call.path(options.urlParams),
  headers: (isLoggedIn() && { Authorization: `JWT ${getToken()}` }),
}).catch(helpers.redirectIfAuthInvalid({ history }));

const apiCreator = ({ history }) => {
  const curriedAxios = prepareAxios({ history });

  /* eslint-disable no-param-reassign */
  const recursivelyAddMethod = (obj = {}, call, currentIndex = 0) => {
    const pathArray = call.name.split('.');
    const currentPath = pathArray[currentIndex];
    if (currentIndex === (pathArray.length - 1)) {
      if (obj[currentPath]) throw new Error(`Call already exists for ${call.name}`);
      obj[currentPath] = curriedAxios(call);
    } else {
      obj[currentPath] = recursivelyAddMethod(obj[currentPath], call, (currentIndex + 1));
    }
    return obj;
  };
  /* eslint-enable */

  const api = callsList.reduce((obj, call, index) => {
    if (!call.name) throw Error(`No name provided for call at index ${index}`);
    return recursivelyAddMethod(obj, call);
  }, {});

  api.helpers = helpers;

  return api;
};

export default apiCreator;
