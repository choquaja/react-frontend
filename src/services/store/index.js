import createBrowserHistory from 'history/createBrowserHistory';
import { createStore, applyMiddleware, compose } from 'redux';
import createLogger from 'redux-logger';
import { createLogicMiddleware } from 'redux-logic';
import rootReducer, { logic } from '../../reducer';
import apiCreator from '../../services/api';
import { extract } from './helpers';

export const INITIAL_STATE = {};
export const history = createBrowserHistory();
export const api = apiCreator({ history });

// Remove trailing slash on paths
history.listen((location) => {
  if (location.pathname.endsWith('/')) history.replace(location.pathname.slice(0, -1));
});

// eslint-disable-next-line no-underscore-dangle
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const isProd = process.env.NODE_ENV === 'production';
const logicMiddleware = createLogicMiddleware(logic, { api, history, extract });
const middlewares = [
  (!isProd && createLogger({
    collapsed: (_, action) => !action.error,
  })),
  logicMiddleware,
].filter(Boolean);

export const store = createStore(
  rootReducer,
  INITIAL_STATE,
  composeEnhancers(applyMiddleware(...middlewares)),
);

export const addLogic = logicMiddleware.addLogic;
