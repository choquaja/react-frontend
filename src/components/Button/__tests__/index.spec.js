import React from 'react';
import { shallow } from 'enzyme';
import Button from '../';
import { theme } from '../../../services/theme';

const required = {
  children: <div>Hello</div>,
  theme,
};
describe('Button', () => {
// Smoke test
  it('Renders without crashing', () => {
    shallow(<Button {...required} />);
  });

// Tests for every prop variation
  it('Renders a block variant of button', () => {
    const button = shallow(<Button {...required} block />);
    expect(button).toMatchSnapshot();
  });

  it('Renders a full button variant', () => {
    const button = shallow(<Button {...required} full />);
    expect(button).toMatchSnapshot();
  });

  it('Renders a large button variant', () => {
    const button = shallow(<Button {...required} large />);
    expect(button).toMatchSnapshot();
  });

  it('Renders a small button variant', () => {
    const button = shallow(<Button {...required} small />);
    expect(button).toMatchSnapshot();
  });

  it('Renders a grouped button variant', () => {
    const button = shallow(<Button {...required} grouped />);
    expect(button).toMatchSnapshot();
  });

  it('Renders a primary button variant', () => {
    const button = shallow(<Button {...required} primary />);
    expect(button).toMatchSnapshot();
  });

  it('Renders a secondary button variant', () => {
    const button = shallow(<Button {...required} secondary />);
    expect(button).toMatchSnapshot();
  });

  it('Renders a success button variant', () => {
    const button = shallow(<Button {...required} success />);
    expect(button).toMatchSnapshot();
  });

  it('Renders a info button variant', () => {
    const button = shallow(<Button {...required} info />);
    expect(button).toMatchSnapshot();
  });

  it('Renders a warning button variant', () => {
    const button = shallow(<Button {...required} warning />);
    expect(button).toMatchSnapshot();
  });

  it('Renders a error button variant', () => {
    const button = shallow(<Button {...required} error />);
    expect(button).toMatchSnapshot();
  });
});
