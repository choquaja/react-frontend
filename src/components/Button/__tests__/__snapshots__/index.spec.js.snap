// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Renders a block variant of button 1`] = `
<button
  className="Button__Button-jkHDFw cKjEvr"
>
  <div>
    Hello
  </div>
</button>
`;

exports[`Button Renders a error button variant 1`] = `
<button
  className="Button__Button-jkHDFw iZudAB"
>
  <div>
    Hello
  </div>
</button>
`;

exports[`Button Renders a full button variant 1`] = `
<button
  className="Button__Button-jkHDFw kIFhOk"
>
  <div>
    Hello
  </div>
</button>
`;

exports[`Button Renders a grouped button variant 1`] = `
<button
  className="Button__Button-jkHDFw gDTZol"
>
  <div>
    Hello
  </div>
</button>
`;

exports[`Button Renders a info button variant 1`] = `
<button
  className="Button__Button-jkHDFw gRFuIj"
>
  <div>
    Hello
  </div>
</button>
`;

exports[`Button Renders a large button variant 1`] = `
<button
  className="Button__Button-jkHDFw hBENRW"
>
  <div>
    Hello
  </div>
</button>
`;

exports[`Button Renders a primary button variant 1`] = `
<button
  className="Button__Button-jkHDFw hStQNt"
>
  <div>
    Hello
  </div>
</button>
`;

exports[`Button Renders a secondary button variant 1`] = `
<button
  className="Button__Button-jkHDFw gkJDPU"
>
  <div>
    Hello
  </div>
</button>
`;

exports[`Button Renders a small button variant 1`] = `
<button
  className="Button__Button-jkHDFw kxAkNX"
>
  <div>
    Hello
  </div>
</button>
`;

exports[`Button Renders a success button variant 1`] = `
<button
  className="Button__Button-jkHDFw caAocj"
>
  <div>
    Hello
  </div>
</button>
`;

exports[`Button Renders a warning button variant 1`] = `
<button
  className="Button__Button-jkHDFw gnGDMo"
>
  <div>
    Hello
  </div>
</button>
`;
